normandy/recipes/tests/test_models.py

```diff
@@ -51,27 +54,39 @@ def test_canonical_json(self):
         recipe = RecipeFactory(
             action=ActionFactory(name='action'),
             arguments_json='{"foo": 1, "bar": 2}',
+            channels=[ChannelFactory(slug='beta')],
+            countries=[CountryFactory(code='CA')],
             enabled=False,
-            filter_expression='2 + 2 == 4',
+            extra_filter_expression='2 + 2 == 4',
+            locales=[LocaleFactory(code='en-US')],
             name='canonical',
         )
         # Yes, this is really ugly, but we really do need to compare an exact
         # byte sequence, since this is used for hashing and signing
+        filter_expression = (
```

Comment:
I don't think this is an appropriate place to test if the filter expression building code, it should have it's own unit tests instead.